### DISCUSSION TITLE
How do I create a player-like entity?

### DISCUSSION THREAD

**Discussion by ryleu** [Minestom/Minestom#1057] [Help] [answered]
I'm trying to make a player-like entity, but I'm not able to directly spawn an entity with the `PLAYER` `EntityType`. Someone on the Discord suggested using a FakePlayer, but doing this doesn't seem to work:
```kt
val newEntity = Entity(EntityType.FakePlayer)
newEntity.isGlowing = true
newEntity.setInstance(overworld!!, Pos(0.0, 10.0, 0.0))
```
I see that there is a `FakePlayer` class, but the constructor seems to be private. How should I go about making this work?

Thanks in advance.

**Comment by GoldenStack** [comment]
Its constructor is protected, so you could try extending it and calling `super`. However, I'm not sure if this is actually how you're supposed to use it.

**Reply by GreatWyrm** [reply]
I believe you use `FakePlayer.initPlayer();`

**Reply by ryleu** [reply]
> I believe you use `FakePlayer.initPlayer();`

that worked! can you make a separate suggested answer so i can mark it correct?

**Comment by KrystilizeNevaDies** [comment]
There should be nothing stopping you from spawning an entity with the player type.

Just note that you will need to also send the player info packet

**Reply by ryleu** [reply]
Ah, that would do it. It wasn't showing up on my client when the code ran. How do I send that packet?

**Reply by amrouda50** [reply]
i also want to know the answer for this question . How do u send the player info packet , or in other worlds where exactly do u send it to ?

**Comment by GreatWyrm** [comment, marked as answer]
You would use [FakePlayer.initPlayer()](https://javadoc.minestom.net/net/minestom/server/entity/fakeplayer/FakePlayer.html#initPlayer(java.util.UUID,java.lang.String,net.minestom.server.entity.fakeplayer.FakePlayerOption,java.util.function.Consumer)) to initialize one of them instead of the constructor.
